vm/src/obj/objcomplex.rs

```diff
@@ -132,6 +140,32 @@ impl PyComplexRef {
         vm.ctx.new_bool(result)
     }
 
+    fn float(self, _vm: &VirtualMachine) -> PyResult {
+        return Err(_vm.new_type_error(String::from("Can't convert complex to float")));
+    }
+
+    fn int(self, _vm: &VirtualMachine) -> PyResult {
+        return Err(_vm.new_type_error(String::from("Can't convert complex to int")));
+    }
+
+    fn mul(self, other: PyObjectRef, vm: &VirtualMachine) -> PyObjectRef {
+        if objtype::isinstance(&other, &vm.ctx.complex_type()) {
+            let other_complex = get_value(&other);
+            vm.ctx.new_complex(Complex64::new(
+                self.value.re * other_complex.re - self.value.im * other_complex.im,
+                self.value.re * other_complex.im + self.value.re * other_complex.im,
+            ))
+        } else if objtype::isinstance(&other, &vm.ctx.int_type()) {
+            let other_int = objint::get_value(&other).to_f64().unwrap();
```

Comment:
After rebasing to #886, we can refine `other` like this:
```
let other_complex = to_complex(other, vm)?;
...
```
Then it will process floats too.